modules/build/src/main/scala/scala/build/CrossSources.scala

```diff
@@ -153,6 +153,29 @@ object CrossSources {
 
     val preprocessedInputFromArgs = value(preprocessSources(inputs.flattened()))
 
+    val preprocessedWithUsingDirs =
+      preprocessedInputFromArgs.filter(p => p.options.exists(_ != BuildOptions.monoid.zero))
+
+    if (preprocessedWithUsingDirs.length > 1)
+      val projectFilePath = inputs.elements.projectSettingsFiles.headOption match
+        case Some(s) => s.path
+        case _       => inputs.workspace / "project.scala"
+
+      preprocessedWithUsingDirs
+        .filter(p => os.Path(p.scopePath.path, os.root) != projectFilePath)
+        .foreach { file =>
+          file.scopePath.root match
+            case Right(base: os.Path) =>
+              val path = os.Path(file.scopePath.path, base)
```

Comment:
I do not like this approach. When we process directives we know their positions, can we just somehow reuse it?

Also following code:

```
// Use Scala 3
//> using scala 3
```

is a valid directive but will not be picked up.

Reply:
I just added new approach, which uses locations extracted from files.